lib/avro_turf.rb

```diff
@@ -43,11 +45,34 @@ def decode(encoded_data, schema_name: nil, namespace: @namespace)
     schema = schema_name && resolve_schema(schema_name, namespace)
     reader = Avro::IO::DatumReader.new(nil, schema)
     dr = Avro::DataFile::Reader.new(io, reader)
+
+    # If no reader schema is specified, we'll try to find one matching the name
+    # of the writer's schema.
+    if schema_name.nil?
+      schema = load_readers_schema(reader.writers_schema)
+      reader.readers_schema = schema if schema
+    end
+
     dr.first
   end
 
   private
 
+  # Returns the reader schema matching the writer's schema if one is locally
+  # available, or nil if there's no matching reader schema.
+  def load_readers_schema(writers_schema)
```

Comment:
This function doesn't specifically load reader schemas- it loads whatever schema you hand it (which may or may not be a readers schema.)  Likewise, the argument doesn't really need to be a writer schema, it can be any schema.  I think the names here are overly specific and misleading.  How about `load_schema` or `load_schema_with_validation`, and `schema_name` instead of `writers_schema`?